distribution/lib/Standard/Base/0.0.0-dev/src/Data/Enso_Cloud/Enso_File.enso

```diff
@@ -0,0 +1,196 @@
+    read : File_Format -> Problem_Behavior -> Any ! Illegal_Argument | File_Error
+    read self format=Auto_Detect (on_problems=Problem_Behavior.Report_Warning) =
+        _ = on_problems
+        case format of
+            Auto_Detect -> case self.asset_type of
+                Enso_Asset_Type.Folder -> self.list
+                Enso_Asset_Type.Project -> Error.throw (Illegal_Argument.Error "Projects cannot be read within Enso code. Open using the IDE.")
+                Enso_Asset_Type.Connection -> Error.throw (Illegal_Argument.Error "Connections cannot be read within Enso code. Open using the IDE.")
+                Enso_Asset_Type.File ->
+                    real_format = Auto_Detect.get_reading_format self
+                    if real_format == Nothing then Error.throw (File_Error.Unsupported_Type self) else
+                        self.read real_format on_problems
+            _ -> self.with_input_stream [File_Access.Read] format.read_stream
+
+    ## ALIAS load bytes, open bytes
+       Reads all bytes in this file into a byte vector.
+    read_bytes : Vector ! File_Error
+    read_bytes self =
+        self.read Bytes
+
+    ## ALIAS load text, open text
+       Reads the whole file into a `Text`, with specified encoding.
+
+       Arguments:
+       - encoding: The text encoding to decode the file with. Defaults to UTF-8.
+       - on_problems: Specifies the behavior when a problem occurs during the
+         function.
+         By default, a warning is issued, but the operation proceeds.
+         If set to `Report_Error`, the operation fails with a dataflow error.
+         If set to `Ignore`, the operation proceeds without errors or warnings.
+    @encoding Encoding.default_widget
+    read_text : Encoding -> Problem_Behavior -> Text ! File_Error
+    read_text self (encoding=Encoding.utf_8) (on_problems=Problem_Behavior.Report_Warning) =
+        self.read (Plain_Text_Format.Plain_Text encoding) on_problems
+
+    ## GROUP Metadata
+       Returns the extension of the file.
+    extension : Text
+    extension self = if self.is_directory then Error.throw (Illegal_Argument.Error "Directories do not have extensions." self.uri) else
+        name = self.name
+        last_dot = name.locate "." mode=Matching_Mode.Last
+        if last_dot.is_nothing then "" else
+            extension = name.drop (Index_Sub_Range.First last_dot.start)
+            if extension == "." then "" else extension
+
+    ## Gets a list of assets within self.
+    list : Vector Enso_File
+    list self = if self.asset_type != Enso_Asset_Type.Folder then Error.throw (Illegal_Argument.Error "Only folders can be listed.") else
+        auth_header = Utils.authorization_header
+        response = HTTP.fetch self.internal_uri HTTP_Method.Get [auth_header]
+        response.if_not_error <|
+            js_object = response.decode_as_json
+            assets = js_object.get "assets" []
+            files = assets.map t-> t:Enso_File
```

Comment:
```suggestion
            files = assets.map t-> t:Enso_File
```

We are using this construct more and more often, I wonder if we should think about adding some support for it, like something similar to `assets.cast Enso_File` (not sure about the name, as clashing with `Table.cast` is not ideal).